lib/sensu/client/process.rb

```diff
@@ -373,6 +373,35 @@ def complete_checks_in_progress
         end
       end
 
+      # Create a deregistration event for a client. Client
+      # definitions may contain `:deregistration` configuration,
+      # containing custom attributes and handler information. By
+      # default, the deregistration event sets the `:handler` to
+      # `registration`. If the client provides its own `:deregistration`
+      # configuration, it's deep merged with the defaults. The
+      # check `:name`, `:output`, `:status`, `:issued`, and
+      # `:executed` values are always overridden to guard against
+      # an invalid definition. The check `:interval` and `:refresh`
+      # values are always overridden to guard against deregistration
+      # events being filtered out by the server.
+      def send_deregister_event
+        check = {:handler => "deregistration"}
+        if @settings[:client].has_key?(:deregistration)
+          check = deep_merge(check, @settings[:client][:deregistration])
+        end
+        timestamp = Time.now.to_i
+        overrides = {
+          :name => "deregistration",
+          :output => "client initiated deregistration",
+          :status => 1,
+          :issued => timestamp,
+          :executed => timestamp,
+          :interval => 1,
+          :refresh => 1
```

Comment:
`refresh` is a custom attribute and can be added/controlled by the `deregistration` merge(). Don't think this should be here.